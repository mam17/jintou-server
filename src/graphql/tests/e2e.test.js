import gql from 'graphql-tag';
import { typeDefs, resolvers, context, ApolloServer } from '../..';

const { createTestClient } = require('apollo-server-testing');

const GET_HOME = gql`
  query {
    home(maxDistance: "8000") {
      name
      _id
      images {
        image
      }
    }
  }
`;

it('fetches single launch', async () => {
  // create a test server to test against, using our production typeDefs,
  // resolvers, and dataSources.
  const server = new ApolloServer({
    typeDefs,
    resolvers,
    dataSources: () => ({}),
    context
  });

  // mock the dataSource's underlying fetch methods
  // launchAPI.get = jest.fn(() => [mockLaunchResponse]);
  // userAPI.store = mockStore;
  // userAPI.store.trips.findAll.mockReturnValueOnce([{ dataValues: { launchId: 1 } }]);

  // use the test server to create a query function
  const { query } = createTestClient(server);

  // run query against the server and snapshot the output
  const res = await query({ query: GET_HOME, variables: { id: 1 } });
  console.log(res.data.home);
  expect(res).toMatchSnapshot();
});
