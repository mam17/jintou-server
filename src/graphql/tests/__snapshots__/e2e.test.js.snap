// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetches single launch 1`] = `
Object {
  "data": Object {
    "home": Array [
      Object {
        "_id": "5e5c795ee512f03030753016",
        "images": Array [
          Object {
            "image": "https://firebasestorage.googleapis.com/v0/b/jintou-d0ad5.appspot.com/o/5e5c795ee512f03030753016.5e5c7b8ce512f0303075301c.png?alt=media&token",
          },
        ],
        "name": "Najila Seabra",
      },
      Object {
        "_id": "5e5c1835792f271026cfc596",
        "images": Array [],
        "name": null,
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
